### PR TITLE
Carrier Hugger Sprites

### PR DESCRIPTION
**Pull request by kugamo**
## About The Pull Request

Carriers' sprites will now only show huggers when you are carrying them. The sprite only has 4 huggers, so each represents a fourth of the total possible amount that can be carried. Dynamically added so their order is not the same everytime.  Also because shaman and eggsac carriers cannot carry huggers, their sprites will now reflect that.

## Why It's Good For The Game

image

https://user-images.githubusercontent.com/16618648/167294652-4fbed4fa-2d13-48af-a589-b9471c2b3c1c.mp4

old MR ---> https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/2759

## Changelog

:cl: Kugamo
imageadd: Carriers' sprites now reflect how many huggers they are currently carrying.
/:cl:

### CODE REVIEW

**Comment by kugamo** [comment]
Old MR had sprites approved by @thwomper . Hopefully that still stands. :0)

**Comment by kugamo** [comment]
makesweet-07dxid

**Comment by Mirvra** [comment]
Wait... Those were huggers this whole time?

Oh wow, had no freaking idea.

**Review by stanalbatross — COMMENTED** [review]
does this work with xenonid carriers (os sprite carriers)

**Comment by kugamo** [comment]
> does this work with xenonid carriers (os sprite carriers)

mutated hive carriers dont get the hugger overlays no, I think corrupted do but I had a hard time testing it.

**Comment by Carnage068** [comment]
> This PR has been inactive for long enough to be automatically marked as stale. This means it is at risk of being auto closed in ~ 7 days, please address any outstanding review items and ensure your PR is finished, if these are all true and you are auto-staled anyway, you need to actively ask maintainers if your PR will be merged. Once you have done any of the previous actions then you should request a maintainer remove the stale label on your PR, to reset the stale timer. If you feel no maintainer will respond in that time, you may wish to close this PR youself, while you seek maintainer comment, as you will then be able to reopen the PR yourself

really

File: code/modules/mob/living/carbon/xenomorph/castes/Carrier.dm [hunk 1]
```diff
@@ -66,6 +68,50 @@
...
 	icon_xenonid = 'icons/mob/xenonids/carrier.dmi'
 
+/mob/living/carbon/Xenomorph/Carrier/update_icons()
+	. = ..()
+
+	overlays.Cut()
+	var/amount_o_huggers = 0
+	if(huggers_max) //no dividing by zero, sorry
+		amount_o_huggers = round(( huggers_cur / huggers_max ) * 100)
+	var/funny_list = list(1,2,3,4)		//all this logic just so that the sprites dont arrive in the same order, for some nice dynamism :0)
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/castes/Carrier.dm:78]:
```suggestion
```
unused in this proc

**Comment by kugamo** [comment]
I feel I should note this will change the default sprites in the codebase to the [redacted] franchise look alike versions, and will necessitate that the ones used on the live code be changed to the huggerless versions (which is outside my control). Otherwise there'll be no visual change on the live game version, since adding or removing hugger sprites from something that already has them wont change how it looks.

**Review by stanalbatross — APPROVED** [review]
@Nanu308 please get these sprites in box too

**Comment by kugamo** [comment]
> @Nanu308 please get these sprites in box too

message me on discord for the dmi if you dont want to look through the commit history. I removed the xeno sprites from this PR, so that the public repo remains clean of them. I'll give you the dmi with the actual xeno sprites for the box then

**Review by nauticall — APPROVED** [review]
sprites look good! good on you, using overlays, makes spriters happy :)

**Pull request merged**